src/Workspaces/Core/Portable/EmbeddedLanguages/Json/JsonLexer.cs

```diff
@@ -0,0 +1,413 @@
+            Position++;
+            return (chars, kind, null);
+        }
+
+        private ImmutableArray<JsonTrivia> ScanTrivia(bool leading)
+        {
+            var result = ArrayBuilder<JsonTrivia>.GetInstance();
+
+            var start = Position;
+
+            while (Position < Text.Length)
+            {
+                var comment = ScanComment();
+                if (comment != null)
+                {
+                    result.Add(comment.Value);
+                    continue;
+                }
+
+                var endOfLine = ScanEndOfLine();
+                if (endOfLine != null)
+                {
+                    result.Add(endOfLine.Value);
+
+                    if (leading)
+                    {
+                        continue;
+                    }
+                }
+
+                var whitespace = ScanWhitespace();
+                if (whitespace != null)
+                {
+                    result.Add(whitespace.Value);
+                    continue;
+                }
+
+                break;
+            }
+
+            return result.ToImmutableAndFree();
+        }
+
+        private JsonTrivia? ScanEndOfLine()
+        {
+            var start = Position;
+            if (IsAt("\r\n"))
+            {
+                Position += 2;
+                return CreateTrivia(JsonKind.EndOfLineTrivia, GetSubPattern(start, Position));
+            }
+            else if (IsAt("\r") || IsAt("\n"))
+            {
+                Position++;
+                return CreateTrivia(JsonKind.EndOfLineTrivia, GetSubPattern(start, Position));
+            }
+
+            return null;
+        }
+
+        public JsonTrivia? ScanComment()
+        {
+            if (IsAt("//"))
+            {
+                return ScanSingleLineComment();
+            }
+            else if (IsAt("/*"))
+            {
+                return ScanMultiLineComment();
+            }
+            else if (IsAt("/"))
+            {
+                var start = Position;
+                Position++;
+
+                var chars = GetSubPattern(start, Position);
+                return CreateTrivia(JsonKind.SingleLineCommentTrivia, chars,
+                    ImmutableArray.Create(new EmbeddedDiagnostic(
+                        WorkspacesResources.Error_parsing_comment,
+                        GetSpan(chars))));
+            }
+
+            return null;
+        }
+
+        private JsonTrivia ScanSingleLineComment()
+        {
+            var start = Position;
+            Position += 2;
+
+            while (Position < Text.Length && 
+                   this.CurrentChar is var ch &&
+                   ch != '\r' && ch != '\n')
+            {
+                Position++;
+            }
+
+            var chars = GetSubPattern(start, Position);
+            if (Position == start + 2)
+            {
+                var diagnostics = ImmutableArray.Create(new EmbeddedDiagnostic(
```

Comment:
What if `//` are the last two characters in the file?

Reply:
then you get an unterminated comment diagnostic.  This matches json.net.  You can see a test for that here: https://github.com/dotnet/roslyn/pull/25521/files#diff-d0475d1c64e0a4bb26665ae88ff3cf24R158

Will add a comment explaining this is intentional.

Reply:
added comment.  #resolved.